File: SyncPad/SyncPad.Shared/Models/FileModels.cs
namespace SyncPad.Shared.Models;

/// <summary>
/// 文件信息 DTO
/// </summary>
public class FileItemDto
{
    public int Id { get; set; }
    public required string FileName { get; set; }
    public long FileSize { get; set; }
    public string? MimeType { get; set; }
    public DateTime UploadedAt { get; set; }
    public DateTime ExpiresAt { get; set; }
    public int PositionX { get; set; } = 0;
    public int PositionY { get; set; } = 0;
}

/// <summary>
/// 文件列表响应
/// </summary>
public class FileListResponse
{
    public List<FileItemDto> Files { get; set; } = [];
}

/// <summary>
/// 文件上传响应
/// </summary>
public class FileUploadResponse
{
    public bool Success { get; set; }
    public FileItemDto? File { get; set; }
    public string? ErrorMessage { get; set; }
}

/// <summary>
/// 文件同步消息（SignalR）
/// </summary>
public class FileSyncMessage
{
    public required string Action { get; set; } // "added", "deleted", "position_changed"
    public FileItemDto? File { get; set; }
    public int? FileId { get; set; } // 删除时使用
}

/// <summary>
/// 文件位置更新请求
/// </summary>
public class FilePositionUpdateRequest
{
    public int FileId { get; set; }
    public int PositionX { get; set; }
    public int PositionY { get; set; }
}

File: SyncPad/SyncPad.Apple/Shared/Models/FileModels.swift
import Foundation

// MARK: - File Item DTO
struct FileItemDto: Codable, Identifiable {
    let id: Int
    let fileName: String
    let fileSize: Int64
    let mimeType: String?
    let uploadedAt: Date
    let expiresAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case fileName
        case fileSize
        case mimeType
        case uploadedAt
        case expiresAt
    }
}

// MARK: - File List Response
struct FileListResponse: Codable {
    let files: [FileItemDto]
}

// MARK: - File Upload Response
struct FileUploadResponse: Codable {
    let success: Bool
    let file: FileItemDto?
    let errorMessage: String?
}

// MARK: - File Sync Message (SignalR)
struct FileSyncMessage: Codable {
    let action: String  // "added", "deleted"
    let file: FileItemDto?
    let fileId: Int?
}

// MARK: - File Status
enum FileStatus {
    case remote  // 仅在服务器
    case cached  // 已缓存
    case error   // 出错
}

// MARK: - File Type Helper
enum FileType: String {
    case image = "图片"
    case pdf = "PDF"
    case document = "文档"
    case spreadsheet = "表格"
    case presentation = "演示"
    case audio = "音频"
    case video = "视频"
    case archive = "压缩包"
    case text = "文本"
    case other = "其他"
}
